### PR TITLE
Feature profile

### LINKED ISSUE #4: User Profile Features
- [ ] my listings button 
- [ ] my offers button 
- [ ] page displays listings by default 
- [ ] listing contains cards within listing info and offer info 
Listing sub section of card: 
- [ ] date posted 
- [ ] listing items 
- [ ] quantity 
- [ ] unit 
Offers sub section of card:
- [ ] date posted 
- [ ] username 
- [ ] listing items 
- [ ] quantity 
- [ ] units 
- [ ] status (reject offer or accept offer buttons)
- my accepted listings section 
- my produce listed and offer <username> produce listed 
- listed preferred contact 
- exchange deadline listed

### PR DESCRIPTION
**Pull request by atownse**
#### What does  this PR do?
Builds out profile components and functions for views

#### Where should the reviewer start?
Start with the profile component and follow imports down to see how components build off each other

#### How should this be manually tested?
Run `npm start` and visit the profile route to examine profile views

#### Any background context you want to provide?
We did not add functionality to accept/decline buttons yet.
Mock data is currently in use

#### What are the relevant tickets?
Ticket #4

### CODE REVIEW

File: src/components/myOffers/MyOffers.js [hunk 1]
```diff
@@ -1,9 +1,90 @@
...
+import mockUserOffers from '../../mockData/mockUserOffer.js'
+
+
+const MyOffers = ({ userListings }) => {
+  
+  const dispatch = useDispatch()
+  
+  useEffect(() => {
+    dispatch(updateUserOffers("userId", mockUserOffers.data.getUserOffers.listings))
+    // console.log(myOffers)
```
Comment (bruce-gordon) [inline src/components/myOffers/MyOffers.js:14]: Let's make sure we remove these console logs and comments

File: src/components/profile/Profile.js [hunk 2]
```diff
@@ -1,9 +1,34 @@
...
+import React, { useState } from 'react';
 import './Profile.scss';
+import MyListings from '../myListings/MyListings.js'
+import MyOffers from '../myOffers/MyOffers.js'
 
 const Profile = () => {
+
+  const [view, setView] = useState(true)
+
+  const handleClick = (e) => {
```
Comment (bruce-gordon) [inline src/components/profile/Profile.js:10]: I feel great about how dry and dynamic we made this!

**Review by bruce-gordon — APPROVED** [review]
Looks good, Aaron!  The only thing I noticed is that Chadrick and I are not being counted as co-authors.  Can we make sure to change that for future commits?

File: src/components/app/App.js [hunk 3]
```diff
@@ -38,6 +39,9 @@ function App() {
         <Route path='/create-new-listing'>
           <NewListingPage />
         </Route>
+        <Route path={'/profile'}>
+          <Profile />
```
Comment (chadrick-d-dev) [inline src/components/app/App.js:43]: Yay! We have a profile route!

File: src/components/profile/Profile.js (hunk 2 above)
Comment (chadrick-d-dev) [reply src/components/profile/Profile.js:10]: Yes, I agree. It is concise!

File: src/mockData/mockUserListings.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+const mockUserListings = {
```
Comment (chadrick-d-dev) [inline src/mockData/mockUserListings.js:1]: We can use this in our tests!

File: src/mockData/mockUserOffer.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
+const mockUserOffers = {
```
Comment (chadrick-d-dev) [inline src/mockData/mockUserOffer.js:1]: We can also use this in our tests!

**Review by chadrick-d-dev — APPROVED** [review]

**Pull request merged**